src/lib/stores/ResultStore.ts

```diff
@@ -0,0 +1,46 @@
+import { get, writable, type Writable } from 'svelte/store';
+
+import type { Stat } from '$lib/types';
+import { toaster } from '$lib/toaster';
+import { branding } from '$lib/configuration';
+import { filters } from '$lib/stores/Filter';
+import { getResultList, promiseList } from '$lib/utilities/StatBuilder';
+import { countResult } from '$lib/utilities/PatientCount';
+import { loadResources } from '$lib/stores/Resources';
+
+export const hasNonZeroResult: Writable<boolean> = writable(false);
+export const resultCounts: Writable<Stat[]> = writable([]);
+export const loading: Writable<Promise<void>> = writable(Promise.resolve());
+export const totalParticipants: Writable<number> = writable(0);
+
+export async function loadPatientCount(isOpenAccess: boolean) {
+  await loadResources();
+  const resultStats: Stat[] = getResultList(isOpenAccess, branding?.results?.stats || []);
+  resultCounts.set(resultStats);
+  loading.set(
+    Promise.allSettled(resultStats.flatMap(promiseList)).then((results) => {
```

Comment:
what happens on an error that isn't handled by rejected? should this be in a try-catch-finally?

Reply:
I can do that - I see no issue with it. I will also do the same for the Landing Stats. Do you prefer a try/catch block here, or (since this function is a promise) to use a catch on the `loadPatientCount` method in the ResultsPanel.svelte page?